FreeRTOS/FreeRTOS
# Fix MingW RTOS demo.

Description
-----------

This PR fixes the WIN32-MingW demo in FreeRTOS/Demo and adds the following changes to functionality:
* Trace recorder continues after saving the Trace output to a file rather than just stopping altogether after a trace dump, allowing users to output / update the trace as many times as they want while the demo runs.
* Keyboard input handling has been moved to a Windows thread and key presses are passed safely into the FreeRTOS simulator using a simulated interrupt.
* Instead of triggering behavior from any key being pressed, you now press 't' to output the trace and you press 'r' to reset the timer for the blinky demo.
* Critical sections have been added around printf calls made by FreeRTOS tasks to prevent a potential deadlock happening if the FreeRTOS scheduler would cause a context switch while a task calls printf. This improves the overall stability of the demo immensely.
Test Steps
-----------

Built and run both the full and blinky demo in using Eclipse IDE 2022-09. Additionally, verified output traces.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
